packages/dataviews/src/dataform-hooks/use-form.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { FormField } from '../types';
+
+export const useForm = < Item >(
+	supportedFields: Record< string, FormField >
+) => {
+	const [ form, setForm ] = useState( {
+		fields: supportedFields,
+		touchedFields: [] as string[],
+		messageErrors: {},
+	} );
```

Comment:
Feels like a bit of a complex state, useReducer might prove to be better in the long term.